src/lib.rs

```diff
@@ -306,24 +309,30 @@ mod test {
     const INET_MTU: usize = 1500;
 
     //  The tests can run in parallel, so make sure to use different ports for all the tests.
-    const fn local_v4(port: u16) -> SocketAddr {
-        SocketAddr::V4(SocketAddrV4::new(Ipv4Addr::LOCALHOST, port))
+    static PORT: AtomicU16 = AtomicU16::new(1234);
+
+    fn new_port() -> u16 {
+        PORT.fetch_add(1, std::sync::atomic::Ordering::Relaxed)
+    }
+
+    fn local_v4() -> SocketAddr {
+        SocketAddr::V4(SocketAddrV4::new(Ipv4Addr::LOCALHOST, new_port()))
     }
 
-    const fn local_v6(port: u16) -> SocketAddr {
-        SocketAddr::V6(SocketAddrV6::new(Ipv6Addr::LOCALHOST, port, 0, 0))
+    fn local_v6() -> SocketAddr {
+        SocketAddr::V6(SocketAddrV6::new(Ipv6Addr::LOCALHOST, new_port(), 0, 0))
     }
 
-    fn inet_v4(port: u16) -> SocketAddr {
-        format!("ietf.org:{port}")
+    fn inet_v4() -> SocketAddr {
+        format!("ietf.org:{}", new_port())
```

Comment:
I wonder if it wouldn't be a good idea to pick a name that doesn't depend on the DNS for this.  Or one that wouldn't require a new lookup.  "github.com", say.

Reply:
I tried hardcoding 1.1.1.1 and its IPv6 equivalent previously, but from the GitHub runner there is no route to these addresses. I wonder if they have some sort of routing setup that requires a DNS lookup.